exist-core/src/main/java/org/exist/xquery/functions/util/Base64Functions.java

```diff
@@ -42,76 +42,80 @@
+                            new FunctionParameterSequenceType("trim", Type.BOOLEAN, Cardinality.EXACTLY_ONE, "Trim trailing newlines?")
+                    },
+                    new FunctionReturnSequenceType(Type.STRING, Cardinality.ZERO_OR_ONE, "the Base64 encoded output"),
+                    "This function is deprecated. The output does not need to be trimmed, please use util:base64-encode#1 instead."
+            ),
+
+            new FunctionSignature(
+                    new QName("base64-decode", UtilModule.NAMESPACE_URI, UtilModule.PREFIX),
+                    "Decode the given Base64 encoded string back to clear text",
+                    new SequenceType[]{
+                            new FunctionParameterSequenceType("string", Type.STRING, Cardinality.ZERO_OR_ONE, "The Base64 string to be decoded")
+                    },
+                    new FunctionReturnSequenceType(Type.STRING, Cardinality.ZERO_OR_ONE, "the decoded output")
+            ),
+
+            new FunctionSignature(
+                    new QName("base64-encode-url-safe", UtilModule.NAMESPACE_URI, UtilModule.PREFIX),
+                    "Encodes the given string as Base64, url-safe. No padding and use - and _ instead of + and /.",
+                    new SequenceType[]{
+                            new FunctionParameterSequenceType("string", Type.STRING, Cardinality.ZERO_OR_ONE, "The string to be Base64 encoded (url-safe)")
+                    },
+                    new FunctionReturnSequenceType(Type.STRING, Cardinality.ZERO_OR_ONE, "the Base64, url-safe encoded output without padding")
+            )
+
     };
-	
 
-    public Base64Functions( XQueryContext context, FunctionSignature signature )
-    {
-        super( context, signature );
+
+    public Base64Functions(XQueryContext context, FunctionSignature signature) {
+        super(context, signature);
     }
-	
-
-    public Sequence eval( Sequence[] args, Sequence contextSequence ) throws XPathException
-    {
-		Sequence value	= Sequence.EMPTY_SEQUENCE;
-		boolean  trim	= true;
-		
-        if( !args[0].isEmpty() ) {       
-			final String str = args[0].getStringValue();
-			
-			if( args.length == 2 ) {
-				trim = args[1].effectiveBooleanValue();
-			}
-	
-	        if (isCalledAs("base64-encode")) {
-				String b64Str = Base64.encodeBase64String(str.getBytes(UTF_8));
-				if (trim) {
-					b64Str = b64Str.trim();
-				}
-				value = new StringValue(b64Str);
-	        } else {
-
-	        	final byte[] data = Base64.decodeBase64(str);
-				value = new StringValue(new String(data, UTF_8));
-	        }
-		}
-		
-		return( value );
+
+
+    public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathException {
+        if (args[0].isEmpty()) {
+            return Sequence.EMPTY_SEQUENCE;
+        }
+
+        final String str = args[0].getStringValue();
+
+        if (isCalledAs("base64-encode")) {
```

Comment:
This would be better as a constant

Reply:
Please explain further what should be a constant here.

Reply:
"base64-encode"